pkg/addons/default/coredns.go

```diff
@@ -1,245 +0,0 @@
-			replicas = int(*coreDNSDeployemnt.Spec.Replicas)
-			image := &coreDNSDeployemnt.Spec.Template.Spec.Containers[0].Image
-			imageParts := strings.Split(*image, ":")
-
-			if len(imageParts) != 2 {
-				return false, fmt.Errorf("unexpected image format %q for %q", *image, KubeProxy)
-			}
-
-			if strings.HasPrefix(imageParts[0], coreDNSImagePrefix) &&
-				strings.HasSuffix(imageParts[0], coreDNSImageSuffix) {
-				*image = coreDNSImagePrefix + region + coreDNSImageSuffix + ":" + imageParts[1]
-			}
-		case "Service":
-			resource.Info.Object.(*corev1.Service).SetResourceVersion(kubeDNSSevice.GetResourceVersion())
-			resource.Info.Object.(*corev1.Service).Spec.ClusterIP = kubeDNSSevice.Spec.ClusterIP
-		}
-
-		status, err := resource.CreateOrReplace(plan)
-		if err != nil {
-			return false, err
-		}
-		logger.Info(status)
-	}
-
-	if waitTimeout != nil && !plan {
-		timer := time.After(*waitTimeout)
-		timeout := false
-		readyPods := sets.NewString()
-		watcher, err := rawClient.ClientSet().CoreV1().Pods(metav1.NamespaceSystem).Watch(listPodsOptions)
-		if err != nil {
-			return false, errors.Wrapf(err, "creating %q pod watcher", CoreDNS)
-		}
-
-		podList, err := rawClient.ClientSet().CoreV1().Pods(metav1.NamespaceSystem).List(listPodsOptions)
-		if err != nil {
-			return false, errors.Wrapf(err, "listing %q pods", CoreDNS)
-		}
-		for _, pod := range podList.Items {
-			if pod.Status.Phase == corev1.PodRunning {
-				readyPods.Insert(pod.Name)
-			}
-		}
-
-		logger.Info("waiting for %d of %q pods to become ready", replicas, CoreDNS)
-		for !timeout && readyPods.Len() < replicas {
-			select {
-			case event := <-watcher.ResultChan():
-				logger.Debug("event = %#v", event)
-				if event.Object != nil && event.Type != watch.Deleted {
-					if pod, ok := event.Object.(*corev1.Pod); ok {
-						if pod.Status.Phase == corev1.PodRunning {
-							readyPods.Insert(pod.Name)
-							logger.Debug("pod %q is ready", pod.Name)
-						} else {
-							logger.Debug("pod %q seen, but not ready yet", pod.Name)
-							logger.Debug("node = %#v", *pod)
-						}
-					}
-				}
-			case <-timer:
-				timeout = true
-			}
-		}
-		watcher.Stop()
-		if timeout {
-			return false, fmt.Errorf("timed out (after %s) waiting for %q pods to become ready", waitTimeout, CoreDNS)
-		}
-	}
-
-	if plan {
-		logger.Info("(plan) would have waited for %q pods to become ready and then delete %q", CoreDNS, KubeDNS)
-		return true, nil
-	}
-
-	if err := rawClient.ClientSet().AppsV1().Deployments(metav1.NamespaceSystem).Delete(KubeDNS, &metav1.DeleteOptions{}); err != nil {
-		return false, errors.Wrapf(err, "deleting %q", KubeDNS)
-	}
-
-	logger.Info("deleted %q", KubeDNS)
-
-	logger.Info("%q is now up-to-date", CoreDNS)
-	return false, nil
-}
-
-// UpdateCoreDNS will update the `coredns` add-on
-func UpdateCoreDNS(rawClient kubernetes.RawClientInterface, region, controlPlaneVersion string, plan bool) (bool, error) {
```

Comment:
We want to keep this function.

Reply:
Change reverted, `InstallCoreDNS` removed and `UpdateCoreDNS` kept.